### PR TITLE
Extend TextLine with TypedSink

### PR DESCRIPTION
**Pull request by snoble**
https://github.com/twitter/scalding/issues/1751

attn @johnynek

### CODE REVIEW

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 1]
```diff
@@ -523,9 +524,12 @@ object TextLine {
     new TextLine(p, sm, textEncoding)
 }
 
-class TextLine(p: String, override val sinkMode: SinkMode, override val textEncoding: String) extends FixedPathSource(p) with TextLineScheme {
+class TextLine(p: String, override val sinkMode: SinkMode, override val textEncoding: String) extends FixedPathSource(p) with TextLineScheme with TypedSink[String] {
   // For some Java interop
+  implicit val tset: TupleSetter[String]
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala]: does this need to be implicit? And can it be `private[this] val...`

**Comment by johnynek** [comment]
can we add a test to https://github.com/twitter/scalding/blob/develop/scalding-core/src/test/scala/com/twitter/scalding/TypedPipeTest.scala. We should use `.runHadoop` I think to test it out.

**Comment by snoble** [comment]
@johnynek test added

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 2]
```diff
@@ -523,9 +523,13 @@ object TextLine {
     new TextLine(p, sm, textEncoding)
 }
 
-class TextLine(p: String, override val sinkMode: SinkMode, override val textEncoding: String) extends FixedPathSource(p) with TextLineScheme {
+class TextLine(p: String, override val sinkMode: SinkMode, override val textEncoding: String)(
+    implicit val tset: TupleSetter[String]
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala]: can we remove this and instead do:
`override def setter[U <: String] = TupleSetter.asSubSetter[String, U](TupleSetter.of[String])` taking the implicit in the constructor changes the binary compatibility without a need that I see.

Comment (snoble) [reply scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala]: oh cool! kk

**Comment by snoble** [comment]
I'll figure out why this test is failing tomorrow

**Comment by johnynek** [comment]
Nice work @snoble ! Thanks for fixing this broken window!

**Pull request merged**